webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -603,6 +388,331 @@ private static void addScrambles(PdfWriter docWriter, Document doc, ScrambleRequ
+        cb.setFontAndSize(bf, fontSize);
+        cb.showTextAligned(PdfContentByte.ALIGN_CENTER, "Graded by: _______________ Result: ______", competitorInfoLeft + (right-competitorInfoLeft)/2, top-offsetTop, 0);
+        offsetTop += fontSize + (marginBottom*(withScrambles ? 1 : 5));
+        cb.endText();
+
+        if (!withScrambles) {
+            cb.beginText();
+            fontSize = 11;
+            cb.setFontAndSize(bf, fontSize);
+            cb.showTextAligned(PdfContentByte.ALIGN_CENTER, "Please see seperate sheet for scrambles.", competitorInfoLeft + (right - competitorInfoLeft) / 2, top - offsetTop, 0);
+            offsetTop += fontSize + marginBottom;
+            cb.endText();
+        }
+
+        cb.beginText();
+        cb.setFontAndSize(bf, 25f);
+        int MAGIC_NUMBER = 40; // kill me now
+        cb.showTextAligned(PdfContentByte.ALIGN_CENTER, "Fewest Moves", left+(competitorInfoLeft-left)/2, top-MAGIC_NUMBER, 0);
+        cb.endText();
+
+        com.itextpdf.text.List rules = new com.itextpdf.text.List(com.itextpdf.text.List.UNORDERED);
+        rules.add("Notate your solution by writing one move per bar.");
+        rules.add("To delete moves, clearly erase/blacken them.");
+        rules.add("Face moves F, B, R, L, U, and D are clockwise.");
+        rules.add("Rotations x, y, and z follow R, U, and F.");
+        rules.add("' inverts a move; 2 doubles a move. (e.g.: U', U2)");
+        rules.add("w makes a face move into two layers. (e.g.: Uw)");
+        rules.add("A [lowercase] move is a cube rotation. (e.g.: [u])");
+
+        ct.addElement(rules);
+        int rulesTop = competitorInfoBottom + (withScrambles ? 55 : 143);
+        ct.setSimpleColumn(left+padding, scrambleBorderTop, competitorInfoLeft-padding, rulesTop, 0, Element.ALIGN_LEFT);
+        ct.go();
+
+        rules = new com.itextpdf.text.List(com.itextpdf.text.List.UNORDERED);
+        rules.add("You have 1 hour to find a solution.");
+        rules.add("Your solution length will be counted in OBTM.");
+        int maxMoves = WCA_MAX_MOVES_FMC;
+        rules.add("Your solution must be at most " + maxMoves + " moves, including rotations.");
+        rules.add("Your solution must not be directly derived from any part of the scrambling algorithm.");
+        ct.addElement(rules);
+        MAGIC_NUMBER = 150; // kill me now
+        ct.setSimpleColumn(left+padding, scrambleBorderTop, rulesRight-padding, rulesTop-MAGIC_NUMBER, 0, Element.ALIGN_LEFT);
+        ct.go();
+
+        doc.newPage();
+    }
+
+    private static void addSolutionSheet(PdfWriter docWriter, Document doc, ScrambleRequest scrambleRequest, String globalTitle) throws DocumentException, IOException {
+        addSolutionSheet(docWriter, doc, scrambleRequest, globalTitle, -1);
+    }
+
+    private static void addScrambleSheet(PdfWriter docWriter, Document doc, ScrambleRequest scrambleRequest, String globalTitle, int index) throws DocumentException, IOException {
```

Comment:
I feel more strongly about adding `Fmc` to the method name here because TNoodle generates lots of scramble sheets, and this is a pretty specific kind of scramble sheet.